influence_benchmark/stats/utils_pandas.py

```diff
@@ -118,6 +118,35 @@ def get_selected_turns_df(turns_df: pd.DataFrame, selected_traj_df: pd.DataFrame
     return pd.merge(turns_df, selected_traj_df, on=["env_name", "initial_state_id", "trajectory_id"])
 
 
+def get_logging_turns_df(
+    turns_df: pd.DataFrame,
+    num_chosen_trajs: int = 1,
+    func: Callable[[pd.DataFrame, int, str], pd.DataFrame] = pd.DataFrame.nlargest,
+    groupby_feats: List[str] = ["env_name"],
+) -> pd.DataFrame:
+    """
+    Get the top-k trajectories based on traj_rew for each environment for a dataframe on the (environment, initial state, trajectory, turn) level. First deduplicate the number of turns.
+
+
+    Args:
+        turns_df (pd.DataFrame): Input DataFrame with entries for each tuple of (environment, initial state, trajectory, turn).
+        num_chosen_trajs (int): Number of top trajectories to select for environment.
+        func (Callable): Function to select top k rows, default is pd.DataFrame.nlargest.
+
+    Returns:
+        pd.DataFrame: DataFrame containing the top-k trajectories for each environment with all of the same columns as the original dataframe.
+    """
+    # Group by env_name, initial_state_id, and trajectory_id, then get the row with max turn
+    df_max_turn = turns_df.loc[turns_df.groupby(["env_name", "initial_state_id", "trajectory_id"])["turn"].idxmax()]
+
+    # Now group by env_name and initial_state_id, and get the top k based on traj_rew
+    top_k_df = (
+        df_max_turn.groupby(groupby_feats).apply(lambda x: func(x, num_chosen_trajs, "traj_rew")).reset_index(drop=True)
+    )
```

Comment:
For an experiment with `max_turns=2`, say that `len(turns_df)=2` (with 1 trajs)

Would `len(top_k_df) = 2` in this case, or `len(top_k_df)=1`? I think it would be len 1, but if that's the case, I think it may break the trajectory logging, because `get_trajs_wandb_html` expects a df which has one row per turn (maybe this should be better documented in it's docstring)

Reply:
I would test this with a max_turns 2 experiment, and check that both steps of the trajectory are being displayed correctly

Reply:
![image](https://github.com/user-attachments/assets/f6a830d7-91af-4a15-865f-a6ce891be12b)

This seems to confirm that you'll only have turn2 in the df after the filtering, which will lead to incomplete logging